app/_src/gateway/upgrade/backup-and-restore.md

```diff
@@ -0,0 +1,157 @@
+
+### Declarative backup
+
+1. To back up data with decK, first make sure it successfully connects to {{site.base_gateway}}:
+
+    ```sh
+    deck ping
+    ```
+
+    If you have RBAC enabled, use the CLI option `--headers` to 
+    specify the admin token. You can specify this token with any decK command:
+
+    ```sh
+    deck ping --headers “Kong-Admin-Token: <password>”
+    ```
+
+2. Use decK to dump the configuration and store the resulting file in a secure location.
+You can back up a particular workspace or all workspaces at once:
+
+    ```sh
+    deck dump --all-workspaces -o /path/to/kong_backup.yaml
+    deck dump --workspace it_dept -o /path/to/kong_backup.yaml
+    ```
+
+    Store the resulting file or files in a safe location.
+
+## Restore Gateway entities
+
+### Database restore
+
+To recover {{site.base_gateway}} configuration data from a database dump, the operator must make sure the database is correctly prepared. 
+For example, for PostgreSQL:
+
+1. In `kong.conf`, set a database user using the `pg_user` parameter:
+
+    ```
+    pg_user = kong
+    ```
+
+2. In `kong.conf`, set a database name using the `pg_database` parameter:
+
+    ```
+    pg_database = kong
+    ```
+
+3. Bootstrap database entities by running `kong migrations bootstrap`.
+
+    Refer to the [`kong migrations` CLI reference](/gateway/{{page.kong_version}}/reference/cli/#kong-migrations) for more information.
+
+4. You can now restore the data using the utility `pg_restore`:
+
+    ```sh
+    pg_restore -U kong -C -d postgres --if-exists --clean kongdb_backup_20230816/
+    ```
+
+### Declarative restore
+
+If you need to roll back, change the DB-less {{site.base_gateway}} instance back to the original version, 
+validate the declarative config, then sync it back with your {{site.base_gateway}} instance.
+
+1. Check that {{site.base_gateway}} is online:
+
+    ```sh
+    deck ping
+    ```
+2. Validate the declarative config:
+
+    ```sh
+    deck validate [--online] -s /path/to/kong_backup.yaml
+    ```
+
+3. Once verified, restore a particular workspace or all workspaces at once:
+
+    ```sh
+    deck sync --all-workspaces -s /path/to/kong_backup.yaml
+    deck sync --workspace it_dept -s /path/to/kong_backup.yaml
+    ```
+
+## Keyring materials backup and restore
+
+If you have enabled keyring and data encryption, you must separately back up and restore keyring materials.
+
+{:.important}
+> **Caution**: Make sure to store the encryption key in a safe place. 
+If the encryption key is lost, you will permanently lose access to the encrypted {{site.base_gateway}} 
+configuration data and there is no other way to recover it.
+
+For technical details, refer to the [manual backup method](/gateway/latest/kong-enterprise/db-encryption/#manual-export-and-manual-import) and the [automatic backup method](/gateway/latest/kong-enterprise/db-encryption/#automatic-backup-and-manual-recovery).
+
+## kong.conf backup
```

Comment:
This section contains backup of files other than `kong.conf`. Should we change the title?